cicecore/cicedynB/dynamics/ice_dyn_evp.F90

```diff
@@ -190,7 +190,10 @@ subroutine evp (dt)
 
       real (kind=dbl_kind), allocatable :: &
          strengthU(:,:,:), & ! strength averaged to U points
-         shrU     (:,:,:), & ! shearU array for gridC
+         divUx    (:,:,:), & ! divU array for gridC, x added to differentiate with divu
```

Comment:
another naming option could be divergU

Reply:
If this is for C-grid only could this be divUc? Or is it for gridcd as well?

Reply:
I was struggling with a name.  I guess we cannot or should not use the divu array that exists in ice_state?  That seems to be related, but I think I don't fully understand.  I will switch usage to divergU.  This comes into play for both C and CD as part of strain_rates_U.

Reply:
For the B-grid divu is basically this quantity divided by the gridcell area. It is computed in deformations. So, it is the average of the four corners as follows:

divu(i,j) = p25*(divune + divunw + divuse + divusw) * tarear(i,j)

So, in deformations_T, we compute divu = divT * tarear.

Definitely confusing usage. Maybe divuT and divuU? Then the divu array is only the divergence at the T points as on the B grid?

Reply:
Hi Tony,

There are two things that can create confusion here. 

In the strain rates subroutines, divU and divT are strain rates times area and refer to the divergence (times area) respectively calculated at the U and T points.  

However, in the deformation subroutines, divu is in fact the real divergence (not multiplied by area...) but the 'u' here refers to the velocity (the divergence of the velocity...). divu here is in fact defined at the T point. I find that it should not be called divu in the deformation subroutines. I would keep divU and divT in the strain rates subroutines (everywhere it refers to strain rates times area) and rename divu in the deformation subroutines to divergence.

Reply:
I'm trying to understand the best way forward.  @JFLemieux73, are you suggesting we change the name of divu in ice_state to divergence?  Then the temporary variables in dynamics and strain_rates in particular would use divU and divT?  The other wrinkle is that the B grid uses divune, divunw, etc and those seem to ultimately form the divu (divergence) term.  So would we also have to change divune to divergencene?  What do others think?  

At present, I have changes in my sandbox that use divergU instead of divUx, but happy to change that again if we can agree on a strategy.

There are several other variables in the code that have similar issues where the "T" value is called xxx but the U, N, and E values are xxxU, xxxN, and xxxE.  There are other variables like strintx that have different units depending on the particular variable name and that is also confusing.   We have not done full clean up of all this stuff even though it's on our list.  I could try to go thru before we merge if that's useful, but that would ultimately touch lots of the code.

Let me know what you think.

Reply:
For now, it's probably better to not change the original divu usage, and for other T variables that don't have T maybe we can say the T is the default.  That's certainly true for all of the thermodynamic variables.  A bigger issue is changing the units (multiplying by area) and using similar names.  If it's local to a subroutine and called out there (e.g. please put units on all of the variable declarations), it's okay.  If there's potential for these different kinds of variables to be confused, then let's name them something else.
So please don't change the old divu to divergence.  Fortran doesn't recognize differences in capitalization, although our convention to use capital U, T, N, E helps with readability.  When things are multiplied by area, you can add an 'a' to the name.   Most of the time, multiplying by area turns a 'density' variable (e.g. thickness = volume per unit area) into a content variable (volume).  I don't know of terms for that, for divergence, but many variables in CICE are per unit area in their default form.

Reply:
It's not consistent throughout the code, but when variables are multiplied by ai in history, we sometimes denote them with _ai.